python/ray/tune/ray_trial_executor.py

```diff
@@ -212,16 +226,47 @@ def _setup_remote_runner(self, trial, reuse_allowed):
             logger.debug("Cannot reuse cached runner {} for new trial".format(
                 self._cached_actor))
             with self._change_working_directory(trial):
-                self._trial_cleanup.add(trial, actor=self._cached_actor)
+                self._trial_cleanup.add(
+                    trial,
+                    actor=self._cached_actor,
+                    placement_group=self._committed_placement_groups.get(
+                        trial, [None])[0])
+                if trial in self._committed_placement_groups:
+                    del self._committed_placement_groups[trial]
             self._cached_actor = None
 
         _actor_cls = _class_cache.get(trial.get_trainable_cls())
-        full_actor_class = _actor_cls.options(
-            num_cpus=trial.resources.cpu,
-            num_gpus=trial.resources.gpu,
-            memory=trial.resources.memory or None,
-            object_store_memory=trial.resources.object_store_memory or None,
-            resources=trial.resources.custom_resources)
+        if isinstance(trial.resources, PlacementGroupFactory):
+            if trial not in self._committed_placement_groups:
+                # This creates the placement group and will try to
+                # allocate resources
+                placement_group = trial.resources()
```

Comment:
I'm thinking maybe it even makes sense for trials always to have a Resource object, and it can have an additional `placement_group` factory that doesn't overload the `resource` object.

You can always generate a Resource from a placement group, right?

Reply:
I guess it does make sense not to overload this. I'll update the code accordingly